### PR TITLE
feat: nativecript 3.1 update with proper scaffolding

### PR DESCRIPTION
**Pull request by vip-git**

### CODE REVIEW

File: src/package.json [hunk 1]
```diff
@@ -40,19 +41,25 @@
   ],
   "author": {
     "name": "Essent",
-    "email": "oscar.lodriguez@essent.nl"
+    "email": "team@essent.nl"
```
Comment (spike1292) [inline src/package.json]: you can use frontend-licenties@essent.nl

File: src/urbanairship.ios.ts [hunk 2]
```diff
@@ -4,6 +4,9 @@ import { UrbanAirshipSettings, CommonUrbanAirship } from './urbanairship.common'
 declare const UANotificationOptionAlert: any;
 declare const UANotificationOptionBadge: any;
 declare const UANotificationOptionSound: any;
+declare const UAPushNotificationDelegate: any;
+declare const UAConfig: any;
+declare const UAirship: any;
```
Comment (spike1292) [inline src/urbanairship.ios.ts]: iOS typings where provided in the old version so the should still be added

`/// <reference path="./objc!AirshipKit.d.ts" />`

File: src/urbanairship.android.ts [hunk 3]
```diff
@@ -1,5 +1,7 @@
 import { UrbanAirshipSettings, CommonUrbanAirship } from './urbanairship.common';
-import * as app from 'application';
+import * as app from "tns-core-modules/application";
+
+declare const com: any;
```
Comment (spike1292) [inline src/urbanairship.android.ts]: android typings are removed. if you added you should not need this.
	
`/// <reference path="./com.urbanairship.d.ts" />`

File: demo/app/tests/tests.js [hunk 4]
```diff
@@ -1,13 +1,12 @@
-var NsUrbanairship = require("nativescript-ns-urbanairship").NsUrbanairship;
-var nsUrbanairship = new NsUrbanairship();
+var MyPlugin = require("nativescript-my-plugin").MyPlugin;
```
Comment (spike1292) [inline demo/app/tests/tests.js]: why is this MyPlugin?

File: demo/app/App_Resources/iOS/demo.entitlements [hunk 5]
```diff
@@ -1,8 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (spike1292) [inline demo/app/App_Resources/iOS/demo.entitlements]: you need this to make push notifications work

File: demo/app/app.ts [hunk 6]
```diff
@@ -1,8 +1,2 @@
-﻿import * as application from 'application';
-import { PushNotification } from './push-notification';
-import { NsUrbanairship } from 'nativescript-urban-airship';
-
-// place your initialize before the start of your application
-PushNotification.initialize();
-
+﻿import * as application from 'tns-core-modules/application';
```
Comment (spike1292) [inline demo/app/app.ts]: where is the urban airship plugin setup?

File: demo/package.json [hunk 7]
```diff
@@ -1,28 +1,30 @@
...
+    "tns-android": {
+      "version": "3.1.1"
     }
-}
\ No newline at end of file
+  },
+  "dependencies": {
+    "nativescript-theme-core": "^1.0.4",
+    "nativescript-unit-test-runner": "^0.3.4",
+    "nativescript-urban-airship": "file:///Users/vipinessent/htdocs/nativescript-plugin-seed/src",
```
Comment (spike1292) [inline demo/package.json]: remove this dep.
if this is here when someone runs setup it will fail

File: src/platforms/ios/Podfile [hunk 8]
```diff
@@ -0,0 +1 @@
+pod 'UrbanAirship-iOS-SDK'
```
Comment (spike1292) [inline src/platforms/ios/Podfile]: missing version number

File: demo/package.json (hunk 7 above)
Comment (vip-git) [inline demo/package.json]: we can ignore this . i have added in readme to run `npm run clean` . It should change back to dev's local instance then.

**Pull request closed without merging**